src/app/component/edit-product/edit-product.component.ts

```diff
@@ -0,0 +1,56 @@
+import { Location } from '@angular/common';
+import { Component } from '@angular/core';
+import { FormBuilder, Validators } from '@angular/forms';
+import { ActivatedRoute } from '@angular/router';
+import { Product } from '../../model/product.model';
+import { ProductService } from '../../service/product.service';
+
+@Component({
+  selector: 'app-edit-product',
+  templateUrl: './edit-product.component.html',
+  styleUrls: ['./edit-product.component.scss']
+})
+export class EditProductComponent {
+  productId: number = 0
+  productName: string = ''
+  productForm = this.fb.group({
+    name: ['', Validators.required],
+    category: ['', Validators.required],
+    image: ['', Validators.required],
+    price: ['', Validators.required],
+    description: ['', Validators.required]
+  })
+
+  constructor(
+    private productService: ProductService,
+    private location: Location,
+    private activatedRoute: ActivatedRoute,
+    private fb: FormBuilder
+  ) {
+    this.activatedRoute.params.subscribe(params =>
+      productService.retrieveProduct(params['id']).subscribe(product => {
+        this.productId = params['id']
+        this.productName = product.name
+        this.productForm.controls['name'].setValue(product.name)
```

Comment:
here if the product definition matches the form definition you can use `this.productForm.patchValue(`